packages/platform-browser/src/dom/dom_renderer.ts

```diff
@@ -508,78 +503,103 @@ function isTemplateNode(node: any): node is HTMLTemplateElement {
 }
 
 class ShadowDomRenderer extends DefaultDomRenderer2 {
-  private shadowRoot: any;
+  private shadowRoot: ShadowRoot;
 
   constructor(
     eventManager: EventManager,
-    private hostEl: any,
-    component: RendererType2,
+    private hostEl: HTMLElement,
+    private component: RendererType2,
     doc: Document,
     ngZone: NgZone,
     nonce: string | null,
     platformIsServer: boolean,
     tracingService: TracingService<TracingSnapshot> | null,
-    private sharedStylesHost?: SharedStylesHost,
+    private sharedStylesHost: SharedStylesHost,
+    private styleScopeService: IsolatedStyleScopeService,
   ) {
     super(eventManager, doc, ngZone, platformIsServer, tracingService);
-    this.shadowRoot = (hostEl as any).attachShadow({mode: 'open'});
 
-    // SharedStylesHost is used to add styles to the shadow root by ShadowDom.
-    // This is optional as it is not used by IsolatedShadowDom.
-    if (this.sharedStylesHost) {
-      this.sharedStylesHost.addHost(this.shadowRoot);
+    // Only create shadow root in browser environments
+    if (!platformIsServer) {
+      this.shadowRoot = (hostEl as HTMLElement).attachShadow({mode: 'open'});
+    } else {
+      // In SSR or environments without shadow DOM support, throw
+      throw new Error(
+        'IsolatedShadowRoot is not supported in SSR mode until declarative shadow DOM is supported.',
+      );
     }
+
+    // Determine if this is isolated based on component encapsulation
+    const isIsolated = component.encapsulation === ViewEncapsulation.IsolatedShadowDom;
+
+    // Register shadow root with StyleScopeService for style targeting (only if service available and in browser)
+    if (this.styleScopeService) {
+      if (isIsolated) {
+        this.styleScopeService.registerIsolatedShadowRoot(this.shadowRoot);
+        this.sharedStylesHost.addShadowRoot(this.shadowRoot);
+      } else {
+        this.styleScopeService.registerStandardShadowRoot(this.shadowRoot);
+        this.sharedStylesHost.addHost(this.shadowRoot);
+      }
+    }
```

Comment:
Nit: I think `styleScopeService` should always be provided and we can probably drop this check?
```suggestion
    if (isIsolated) {
      this.styleScopeService.registerIsolatedShadowRoot(this.shadowRoot);
      this.sharedStylesHost.addShadowRoot(this.shadowRoot);
    } else {
      this.styleScopeService.registerStandardShadowRoot(this.shadowRoot);
      this.sharedStylesHost.addHost(this.shadowRoot);
    }
```